src/Views/ProgressView.vala

```diff
@@ -112,6 +112,27 @@ public class ProgressView : AbstractInstallerView {
         return terminal_view.buffer.text;
     }
 
+    /**
+     * Device paths in `/dev/mapper/` have special rules for handling the `-` character. This
+     * character is used to differentiate between the volume group identifier, and the
+     * logical volume indentifier in the path name.
+     *
+     * If either identifiers have the `-` character in their name, the correct device path
+     * to reach that device in the file system will need to repeat the `-` character a
+     * second time. The device will continue to be internally referenced by lvm tools and
+     * distinst as its original name, though.
+     *
+     * Note: the device path specified here should refer to a LVM or LUKS device, rather
+     * than a logical volume that exists on a LVM device.
+     */
+    private string extract_volume_group_from (string device_path) {
+        return device_path
+            // Remove the `/dev/mapper` prefix
+            .offset (12)
+            // Replace all occurrences of `--` with `-`
```

Comment:
This still just says the what (which is obvious from the line below), and not the why. The description about the special treatment of `-` would make sense in this comment.

Reply:
The why is explained in the function's description.

Reply:
Ah, I see that now. 👍